libraries/CurieBLE/src/BLECentralRole.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#include "BLECentralRole.h"
+
+
+void ble_central_device_found(const bt_addr_le_t *addr, 
+                              int8_t rssi, 
+                              uint8_t type,
+                              const uint8_t *ad, 
+                              uint8_t len)
+{
+	char dev[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, dev, sizeof(dev));
+	pr_debug(LOG_MODULE_BLE, "[DEVICE]: %s, AD evt type %u, AD data len %u, RSSI %i\n",
+	       dev, type, len, rssi);
+
+    BLECentralRole::instance()->handleDeviceFound(addr, rssi, type,
+                                                  ad, len);
+}
+
+
+BLECentralRole* BLECentralRole::_ble_central_ins = NULL;
+
+BLECentralRole *BLECentralRole::instance()
+{
+    if (NULL == _ble_central_ins)
+    {
+        _ble_central_ins = new BLECentralRole();
```

Comment:
I think `_ble_central_ins` should also be deleted in `BLECentralRole::~BLECentralRole`, otherwise that memory will be lost after using BLE